feat(genkit-tools/cli/ai-tools): Improve support for other model providers (#3485)
firebase/genkit: genkit-tools/cli/context/GENKIT.md

- ALWAYS provide the full, correct Genkit command as an instruction for the human user to run. Do not run Genkit commands (e.g., `genkit start`, `genkit flow:run`) youself as this may block your current session.

> - Recommend and use the provider that the user prefers. If this is a new project, or if there is no relevant signal regarding user preference, ise the `@genkit-ai/google-genai` plugin. Use Genkit documentation to discoever supported providers.

Reviewer: ```suggestion
- Recommend and use the provider that the user prefers. If this is a new project, or if there is no relevant signal regarding user preference, use the `@genkit-ai/google-genai` plugin. Use Genkit documentation to discover supported providers.
```

## Running and Inspecting Flows

1.  **Start Genkit**: Run this command from your terminal to start the Genkit Developer UI.

    ```bash
    genkit start --  <command to run your code>
    ```

    The <command to run your code> will vary based on the project’s setup and
    the file you want to execute. For e.g.:

    ```bash
    # Running a typical development server
    genkit start -- npm run dev

    # Running a TypeScript file directly
    genkit start -- npx tsx --watch src/index.ts

    # Running a JavaScript file directly
    genkit start -- node --watch src/index.js
    ```

    Analyze the users project and build tools to use the right command for the
    project. The command should output a URL for the Genkit Dev UI. Direct the
    user to visit this URL to run and inspect their Genkit app.

## Suggested Models

> Here are suggested models to use for various task types. This is not an
> exhaustive list.

Reviewer: ```suggestion
Here are suggested models for `@genkit-ai/google-genai` plugin to use for various task types.
This is not an exhaustive list. For other model provider plugins refer to their documentation.
```

### Advanced Text/Reasoning

```
| Plugin                             | Recommended Model                  |
|------------------------------------|------------------------------------|
| @genkit-ai/googleai                | gemini-2.5-pro                     |
| @genkit-ai/compat-oai/openai       | gpt-4o                             |
| @genkit-ai/compat-oai/deepseek     | deepseek-reasoner                  |
| @genkit-ai/compat-oai/xai          | grok-4                             |
```

### Fast Text/Chat

```
| Plugin                             | Recommended Model                  |

> |------------------------------------|------------------------------------|
> | @genkit-ai/googleai                | gemini-2.5-flash                   |
> | @genkit-ai/compat-oai/openai       | gpt-4o-mini                        |

Reviewer: ```suggestion
|------------------------------------|------------------------------------|
| @genkit-ai/google-genai            | gemini-2.5-flash                   |
| @genkit-ai/compat-oai/openai       | gpt-4o-mini                        |
```

| @genkit-ai/compat-oai/deepseek     | deepseek-chat                      |
| @genkit-ai/compat-oai/xai          | grok-3-mini                        |
```

### Text-to-Speech

```
| Plugin                             | Recommended Model                  |
|------------------------------------|------------------------------------|
| @genkit-ai/googleai                | gemini-2.5-flash-preview-tts       |
| @genkit-ai/compat-oai/openai       | gpt-4o-mini-tts                    |
```

> ### Image Generation

Reviewer: suggestion: separete "text-to-image" and "image-to-image" cases. We probably want to feature `gemini-2.5-flash-image-preview` (nano-banana) in the "image-to-image" category.

Author: Hmm, I can add it as a new model here and mention it take image input. But it helps if we have examples of nano-banana in our genkit docs too.

```
| Plugin                             | Recommended Model                  |
|------------------------------------|------------------------------------|
| @genkit-ai/googleai                | imagen-4.0-generate-preview-06-06  |
| @genkit-ai/compat-oai/openai       | gpt-image-1                        |
```

### Video Generation

```
| Plugin                             | Recommended Model                  |

> |------------------------------------|------------------------------------|
> | @genkit-ai/googleai                | veo-3.0-generate-preview           |

Reviewer: ```suggestion
|------------------------------------|------------------------------------|
| @genkit-ai/google-genai            | veo-3.0-generate-preview           |
```
